tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -275,6 +275,59 @@ class UpgradeIntegrationTest extends AkkaIntegrationTest with MesosClusterTest w
     marathonCurrent.close()
   }
 
+  "resident app can be restarted after upgrade from 1.6.322" in {
+    val zkUrl = s"$zkURLBase-resident-apps"
+    val marathon16322 = Marathon16322(marathon16322Artifact.marathonPackage, suiteName = s"$suiteName-1-6-322", mesosMasterUrl, zkUrl)
+
+    // Start apps in 1.6.322
+    Given("A Marathon 1.6.322 is running")
+    marathon16322.start().futureValue
+    (marathon16322.client.info.entityJson \ "version").as[String] should be("1.6.322")
+
+    And("new running apps in Marathon 1.6.322")
+    val containerPath = "persistent-volume"
+    val residentApp_16322 = residentApp(
+      id = testBasePath / "resident-app-16322",
+      containerPath = containerPath,
+      cmd = s"""echo "data" >> $containerPath/data && sleep 1000""")
+    marathon16322.client.createAppV2(residentApp_16322) should be(Created)
+
+    patienceConfig
+    eventually { marathon16322 should have (runningTasksFor(residentApp_16322.id.toPath, 1)) }
+    val originalApp16322Tasks = marathon16322.client.tasks(residentApp_16322.id.toPath).value
+
+    When("We restart the app")
```

Comment:
Does restarting an app trigger some failure scenario?

Reply:
Yes. This triggered the same behaviour we saw in the issue.

Reply:
The restart **after** update triggered the issue. This one is obsolete